<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>svg_plotter</class>
 <widget class="QWidget" name="svg_plotter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1198</width>
    <height>831</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QFrame" name="frame">
     <property name="maximumSize">
      <size>
       <width>300</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="load_btn">
        <property name="text">
         <string>Load SVG</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>lsteps:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="lsteps_spin">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Machine</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Width (mm)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Height (mm)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spin_width">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>500</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spin_height">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>500</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>719</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="plot" native="true"/>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>load_btn</sender>
   <signal>clicked()</signal>
   <receiver>svg_plotter</receiver>
   <slot>load_svg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lsteps_spin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>svg_plotter</receiver>
   <slot>compute_paths()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>svg_plotter</receiver>
   <slot>repaint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_height</sender>
   <signal>valueChanged(int)</signal>
   <receiver>svg_plotter</receiver>
   <slot>repaint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>load_svg()</slot>
  <slot>compute_paths()</slot>
 </slots>
</ui>
